tests/lib/rules/new-ember-data-packages.js

```diff
@@ -0,0 +1,185 @@
+          type: 'Identifier',
+        },
+      ],
+    },
+    {
+      code: `import DS from 'ember-data';
+        const { Model } = DS;
+
+        export default Model.extend({
+          name: DS.attr('string')
+        });
+      `,
+      parserOptions,
+      errors: [
+        {
+          message:
+            'Imports from new @ember-data packages should be prefered over imports from ember-data',
+          type: 'ImportDeclaration',
+        },
+        {
+          message: "Use `import Model from '@ember-data/model';` instead of using DS destructuring",
+          type: 'Property',
+        },
+        {
+          message: "Use `import { attr } from '@ember-data/model';` instead of using DS.attr",
+          type: 'Identifier',
+        },
+      ],
+    },
+    {
+      code: `import Model from 'ember-data/model';
+
+        export default Model.extend({
+        });
+      `,
+      parserOptions,
+      errors: [
+        {
+          message:
+            'Imports from new @ember-data packages should be prefered over imports from ember-data',
+          type: 'ImportDeclaration',
+        },
+      ],
+    },
+    {
+      code: `import Model from 'ember-data/model';
+        import attr  from 'ember-data/attr';
+
+        export default Model.extend({
+          name: attr('string')
+        });
+      `,
+      parserOptions,
+      errors: [
+        {
+          message:
+            'Imports from new @ember-data packages should be prefered over imports from ember-data',
+          type: 'ImportDeclaration',
+        },
+        {
+          message:
+            'Imports from new @ember-data packages should be prefered over imports from ember-data',
+          type: 'ImportDeclaration',
+        },
+      ],
+    },
+    {
+      code: `import Model from '@ember-data/model';
+        import attr from 'ember-data/attr';
+
+        export default Model.extend({
+          name: attr('string')
+        });
+      `,
+      parserOptions,
+      errors: [
+        {
+          message:
+            'Imports from new @ember-data packages should be prefered over imports from ember-data',
+          type: 'ImportDeclaration',
+        },
+      ],
+    },
+    {
+      code: `import Model from '@ember-data/model';
+        import namedAttr from 'ember-data/attr';
+
+        export default Model.extend({
+          name: attr('string')
+        });
+      `,
+      parserOptions,
+      errors: [
+        {
+          message:
+            'Imports from new @ember-data packages should be prefered over imports from ember-data',
+          type: 'ImportDeclaration',
+        },
+      ],
+    },
+    {
+      code: `import { Model } from 'ember-data';
+
+        export default Model.extend({
+        });
+      `,
+      parserOptions,
+      errors: [
+        {
+          message:
+            'Imports from new @ember-data packages should be prefered over imports from ember-data',
```

Comment:
likely should drop use of "new" here as well

whew, sorry for like spamming you with a dozen of the same feedback

Reply:
ahah no worries :)

Reply:
I've fixed them